source/npm/util.js

```diff
@@ -100,12 +102,65 @@ exports.verifyRecentNpmVersion = async () => {
 };
 
 exports.checkIgnoreStrategy = ({files}) => {
-	const rootDir = pkgDir.sync();
-	const npmignoreExists = fs.existsSync(path.resolve(rootDir, '.npmignore'));
-
-	if (!files && !npmignoreExists) {
+	if (!files && !npmignoreExistsInPackageRootDir()) {
 		console.log(`
 		\n${chalk.bold.yellow('Warning:')} No ${chalk.bold.cyan('files')} field specified in ${chalk.bold.magenta('package.json')} nor is a ${chalk.bold.magenta('.npmignore')} file present. Having one of those will prevent you from accidentally publishing development-specific files along with your package's source code to npm. 
 		`);
 	}
 };
+
+// Get all files which will be ignored by either `.npmignore` or the `files` property in `package.json` (if defined)
+exports.getNewAndUnpublishedFiles = async (newFiles, globArrayFromFilesProperty) => {
```

Comment:
Maybe define default values for the arguments here so that we won't be dealing with any `undefined` arguments?

Reply:
I added a default value for parameter `newFiles`. `globArrayFromFilesProperty` should be undefined, if missing in `package.json`.